docs/source/format/IPC.rst

```diff
@@ -104,7 +104,7 @@ Schematically we have: ::
 
     <magic number "ARROW1">
     <empty padding bytes [to 8 byte boundary]>
-    <STREAMING FORMAT>
+    <STREAMING FORMAT with EOS>
```

Comment:
Someone should check if Java and JavaScript write the EOS

Reply:
Had a look, it doesn't appear so https://github.com/apache/arrow/blob/master/java/vector/src/main/java/org/apache/arrow/vector/ipc/ArrowFileWriter.java#L67

Reply:
```         return this
+            ._writePadding(4); // eos bytes
             ._write(buffer) // Write the flatbuffer
             ._write(Int32Array.of(buffer.byteLength)) // then the footer size suffix
             ._writeMagic(); // then the magic suffix
```
and
```   protected void endInternal(WriteChannel out) throws IOException {
+    out.writeIntLittleEndian(0); // Write an EOS identifier for sequential readers
     long footerStart = out.getCurrentPosition();
```
? ...but I haven't figured out the testing yet